Setup webpack

bumsyalao opened the pull request:
#### What does this PR do?
- Setup Webpack

#### Description of Task to be completed?
- Setup webpack

#### How should this be manually tested?
n/a
#### Any background context you want to provide?
n/a
#### What are the relevant pivotal tracker stories?
n/a
#### Screenshots (if appropriate)
n/a
#### Questions:
n/a

bumsyalao committed:
chore: setup babel-preset

- install dependencies
- specify babel preset
- add start script
- create directories

[Delivers #1]

bumsyalao committed:
chore: ignore redundant files

- delete user generated files
- Ignore redundant files in gitignore

bumsyalao committed:
chore: setup express server

- configure port for server
- setup webpack copiler file

bumsyalao committed:
chore (setup webpack)

- set webpack config
- Render message

[Delivers #2]

BolajiOlajide commented on .babelrc:
```diff
@@ -0,0 +1,6 @@
+    "react",
+    "es2015"
+  ]
+}
```
## Nitpick

You see the red warning sign here, it means you need to add a new line to this file. You should always add a new line at the end of every file.

BolajiOlajide commented on client/index.html:
```diff
@@ -0,0 +1,8 @@
+    <title> TO DO APP </title>
+  </head>
+  <body>
+    <h1> Hi I'm working </h1>
```
## Critical

So the application isn't working yet, what you should have here should be the `div` we are targeting. Take a look at this: [https://github.com/BolajiOlajide/Webpack3-React-StarterKit/blob/master/src/index.html#L7](https://github.com/BolajiOlajide/Webpack3-React-StarterKit/blob/master/src/index.html#L7).

BolajiOlajide replied:
If you don't understand what I mean, let me know so we can look into it.

BolajiOlajide commented on client/index.html:
```diff
@@ -0,0 +1,8 @@
+  <body>
+    <h1> Hi I'm working </h1>
+  </body>
+</html>
```
## Nitpick

New line here. Because of this red warning

BolajiOlajide commented on package.json:
```diff
@@ -4,6 +4,8 @@
   "description": "A to do application",
   "main": "index.js",
```
## Nitpick

Let this point to the actual entry point we have defined. Since there is no file called `index.js` in the root, this can be a bit misleading.

BolajiOlajide commented on package.json:
```diff
@@ -4,6 +4,8 @@
   "scripts": {
+    "prestart": "babel server.js --out-file server-transpiled.js",
+    "start": "node server-transpiled.js",
     "test": "echo \"Error: no test specified\" && exit 1"
```
## Nitpick

Remove this test script since we are not using it

BolajiOlajide commented on server.js:
```diff
@@ -0,0 +1,31 @@
+const dotenv = require("dotenv");
+const config = require("./webpack.config");
+
+/* eslint-disable no-console */
```
## Nitpick

We are not making use of this line, let's remove it.

BolajiOlajide commented on server.js:
```diff
@@ -0,0 +1,31 @@
+const express = require("express");
```
## Critical

I totally forgot this but since we are transpiling the `server.js` down to ES5 then let's used `ES6` styled imports.

So let's have something like:
```
import express from 'express';
```

instead of what we have there.

BolajiOlajide commented on server.js:
```diff
@@ -0,0 +1,31 @@
+});
+
+app.listen(port, err =>{
+  if (!err) open(`http://localhost:${port}`);  else {
```
## Nitpick

Have you heard of `tenary` operators, if you can make use of it, then it'll be great. If not, you can just skip it and add it to the list of things we'll talk about during our next session.

bumsyalao committed:
chore(Implement requested changes)

- add end of line to babelrc
- specify main entry file
- import files in es6
- specify script source in index.html

bumsyalao committed:
chore: Implement feedback

- set up index.html to render app source
